src/chainparams.cpp

```diff
@@ -461,11 +461,8 @@ void CRegTestParams::UpdateActivationParametersFromArgs(const ArgsManager& args)
 {
     if (args.IsArgSet("-segwitheight")) {
         int64_t height = args.GetArg("-segwitheight", consensus.SegwitHeight);
-        if (height < -1 || height >= std::numeric_limits<int>::max()) {
+        if (height < 0 || height >= std::numeric_limits<int>::max()) {
```

Comment:
Can you commit-split the changes around `segwithheight` from the ones around `NODE_WITNESS` ? Better to review any changes in net_processing on their own.

Reply:
Done - moved the second commit to #21090 and split it up into several there.